src/libs/auth.ts

```diff
@@ -109,20 +120,34 @@ const getSigninArgs = (includeBillingScope) => {
 
 export const signIn = async (includeBillingScope = false) => {
   const args = getSigninArgs(includeBillingScope);
-  const user = await getAuthInstance().signinPopup(args);
-
+  const user: User = await getAuthInstance().signinPopup(args);
+  const generatedUuid = uuid();
+  const sessionStartTime = Date.now();
+  authStore.update((state) => ({
+    ...state,
+    sessionId: generatedUuid,
+    sessionStartTime,
+  }));
   // For B2C record in the auth store whether we requested the GCP cloud-billing scope since there
   // is no way to determine it after the fact.
   // For Google we don't need to do this since we can inspect the scope directly in the user object.
   if (!isGoogleAuthority()) {
     authStore.update((state) => ({ ...state, hasGcpBillingScopeThroughB2C: includeBillingScope }));
   }
-  Ajax().Metrics.captureEvent(Events.userLogin, { authProvider: user.profile.idp });
+  Ajax({}).Metrics.captureEvent(Events.userLogin, {
+    authProvider: user.profile.idp,
+    sessionId: generatedUuid,
+    sessionStartTime: Utils.makeCompleteDate(sessionStartTime),
+  });
   return user;
 };
 
 export const reloadAuthToken = (includeBillingScope = false) => {
   const args = getSigninArgs(includeBillingScope);
+  // session identifier would be ideal to log here
+  Ajax({}).Metrics.captureEvent(Events.userAuthTokenReload, {
```

Comment:
Would it be possible to record whether the reload succeeded or not?

Reply:
Good call, adding it in the finally check of the `getAuthInstance` chain